### PR TITLE
fix unit failures on debug without symfpu

### PR DESCRIPTION
**Pull request by yoni206**
Co-authored-by: Aina Niemetz <aina.niemetz@gmail.com>

The following tests fail (locally) when compiling with debug and without symfpu:
```
unit/api/sort_black (Failed)
unit/theory/logic_info_white (Failed)
unit/util/floatingpoint_black (Child aborted)
```

This PR conditions the relevant parts of these tests to the availability of symfpu.

(ignoring whitespaces is recommended)

### CODE REVIEW

**Review by aniemetz — CHANGES_REQUESTED** [review]
For ` test/unit/util/floatingpoint_black.h `, use the guard (to disable everything if undefined)
```
#ifdef CVC4_USE_SYMFPU
....
#endif
```
Then it's not necessary to guard every single function.

File: test/unit/theory/logic_info_white.h [hunk 1]
```diff
@@ -533,46 +533,47 @@ class LogicInfoWhite : public CxxTest::TestSuite {
...
-    info.disableTheory(THEORY_FP);
-    info.disableTheory(THEORY_SEP);
-    info.disableTheory(THEORY_BAGS);
-    info.lock();
-    TS_ASSERT_EQUALS( info.getLogicString(), "QF_AX" );
-    TS_ASSERT( info.isPure( THEORY_ARRAYS ) );
-    TS_ASSERT( ! info.isQuantified() );
-
+    if (CVC4::Configuration::isBuiltWithSymFPU())
+    {
```
Comment (aniemetz) [inline test/unit/theory/logic_info_white.h:537]:
```suggestion
```

File: test/unit/theory/logic_info_white.h [hunk 2]
```diff
@@ -533,46 +533,47 @@ class LogicInfoWhite : public CxxTest::TestSuite {
...
+      info = info.getUnlockedCopy();
+      TS_ASSERT(!info.isLocked());
+      info.disableTheory(THEORY_STRINGS);
+      info.disableTheory(THEORY_SETS);
+      info.disableTheory(THEORY_BAGS);
+      info.arithOnlyLinear();
+      info.disableIntegers();
+      info.lock();
+      TS_ASSERT_EQUALS(info.getLogicString(), "SEP_AUFBVFPDTLRA");
```
Comment (aniemetz) [inline test/unit/theory/logic_info_white.h:546]:
```suggestion
     if (CVC4::Configuration::isBuiltWithSymFPU())
     {
       TS_ASSERT_EQUALS(info.getLogicString(), "SEP_AUFBVFPDTLRA");
     }
     else
     {
       TS_ASSERT_EQUALS(info.getLogicString(), "SEP_AUFBVDTLRA");
     }

File: test/unit/theory/logic_info_white.h [hunk 3]
```diff
@@ -533,46 +533,47 @@ class LogicInfoWhite : public CxxTest::TestSuite {
...
+      info.disableTheory(THEORY_ARITH);
+      info.disableTheory(THEORY_UF);
+      info.disableTheory(THEORY_FP);
+      info.disableTheory(THEORY_SEP);
+      info.disableTheory(THEORY_BAGS);
+      info.lock();
+      TS_ASSERT_EQUALS(info.getLogicString(), "QF_AX");
+      TS_ASSERT(info.isPure(THEORY_ARRAYS));
+      TS_ASSERT(!info.isQuantified());
+    }
```
Comment (aniemetz) [inline test/unit/theory/logic_info_white.h:576]:
```suggestion
```

File: test/unit/theory/logic_info_white.h [hunk 4]
```diff
@@ -533,46 +533,47 @@ class LogicInfoWhite : public CxxTest::TestSuite {
...
+      info.disableIntegers();
+      info.lock();
+      TS_ASSERT_EQUALS(info.getLogicString(), "SEP_AUFBVFPDTLRA");
+
+      info = info.getUnlockedCopy();
+      TS_ASSERT(!info.isLocked());
+      info.disableQuantifiers();
+      info.disableTheory(THEORY_BAGS);
+      info.lock();
+      TS_ASSERT_EQUALS(info.getLogicString(), "QF_SEP_AUFBVFPDTLRA");
```
Comment (aniemetz) [inline test/unit/theory/logic_info_white.h:553]:
```suggestion
     if (CVC4::Configuration::isBuiltWithSymFPU())
     {
       TS_ASSERT_EQUALS(info.getLogicString(), "QF_SEP_AUFBVFPDTLRA");
     }
     else
     {
       TS_ASSERT_EQUALS(info.getLogicString(), "QF_SEP_AUFBVDTLRA");
     }
```

File: test/unit/theory/logic_info_white.h [hunk 5]
```diff
@@ -533,46 +533,47 @@ class LogicInfoWhite : public CxxTest::TestSuite {
...
+
+      info = info.getUnlockedCopy();
+      TS_ASSERT(!info.isLocked());
+      info.disableTheory(THEORY_BV);
+      info.disableTheory(THEORY_DATATYPES);
+      info.disableTheory(THEORY_BAGS);
+      info.enableIntegers();
+      info.disableReals();
+      info.lock();
+      TS_ASSERT_EQUALS(info.getLogicString(), "QF_SEP_AUFFPLIA");
```
Comment (aniemetz) [inline test/unit/theory/logic_info_white.h:563]:
```suggestion
     if (CVC4::Configuration::isBuiltWithSymFPU())
     {
       TS_ASSERT_EQUALS(info.getLogicString(), "QF_SEP_AUFFPLIA");
     }
     else
     {
       TS_ASSERT_EQUALS(info.getLogicString(), "QF_SEP_AUFLIA");
     }
```

**Review by aniemetz — CHANGES_REQUESTED** [review]

**Review by martin-cs — APPROVED** [review]

**Comment by yoni206** [comment]
OK I think I included your suggestions and addressed other comments (thanks also for the offline help). Please let me know if anything else is needed.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**